### ISSUE TITLE
Total number of samples is shown in the log instead of total number of tokens

### ISSUE THREAD

**Issue by seungduk-yanolja** [axolotl-ai-cloud/axolotl#2334]
### Please check that this issue hasn't been reported before.

- [x] I searched previous [Bug Reports](https://github.com/axolotl-ai-cloud/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior
```
[2025-02-15 16:54:21,828] [DEBUG] [axolotl.calculate_total_num_steps:403] [PID:3264965] [RANK:0] total_num_tokens: 16_452_017_178
[2025-02-15 17:11:49,558] [DEBUG] [axolotl.calculate_total_num_steps:421] [PID:3264965] [RANK:0] `total_supervised_tokens: 16_452_017_178`
```
### Current behaviour
```
[2025-02-15 16:54:21,828] [DEBUG] [axolotl.calculate_total_num_steps:403] [PID:3264965] [RANK:0] total_num_tokens: 31_666_866
[2025-02-15 17:11:49,558] [DEBUG] [axolotl.calculate_total_num_steps:421] [PID:3264965] [RANK:0] `total_supervised_tokens: 16_452_017_178`
```
### Steps to reproduce

Train a model using Axolotl

### Config yaml

```yaml
base_model: Qwen/Qwen2.5-7B
model_type: AutoModelForCausalLM
tokenizer_type: AutoTokenizer

load_in_8bit: false
strict: false

datasets:
  - path: /data/nas-2/seungduk/sanitized.real.final.lt2048.jsonl
    ds_type: json
    type: completion

val_set_size: 0
shuffle_merged_datasets: false
save_safetensors: true

load_in_4bit: true
adapter: qlora
lora_r: 64
lora_alpha: 128
lora_dropout: 0.05
lora_target_linear: true
lora_modules_to_save:
  - embed_tokens
  - lm_head

sequence_len: 2048
sample_packing: true
eval_sample_packing: false
pad_to_sequence_len: true

gradient_accumulation_steps: 4
micro_batch_size: 8
num_epochs: 1
optimizer: adamw_bnb_8bit
lr_scheduler: cosine
learning_rate: 3e-5
cosine_min_lr_ratio: 0.1
use_tensorboard: true

train_on_inputs: false
group_by_length: false
bf16: auto
fp16:
tf32: false

gradient_checkpointing: unsloth
early_stopping_patience:
resume_from_checkpoint:
local_rank:
logging_steps: 1
xformers_attention:
flash_attention: true

warmup_ratio: 0.1
evals_per_epoch: 1
save_steps: 500
save_total_limit: 1000
debug:
deepspeed: /data/nas-2/seungduk/axolotl/deepspeed_configs/zero1_torch_compile.json
weight_decay: 0.1
special_tokens:
```

### Possible solution

DeepSeek said
```
# This version counts SAMPLES because:
total_num_tokens = np.sum(
    train_dataset.select_columns("input_ids")  # Returns Dataset with only input_ids column
    .to_pandas()                              # Creates DF with 1 column ("input_ids")
    .apply(lambda x: len(x))                  # x is a ROW (always length=1)
    .values
)

# While this version counts TOKENS correctly:
total_num_tokens = np.sum(
    train_dataset.data.column("input_ids")     # Directly accesses the input_ids array
    .to_pandas()                              # Creates Series of token lists
    .apply(lambda x: len(x))                  # x is token list → counts tokens
    .values
)
```

and suggested

```
        total_num_tokens = np.sum(
            train_dataset.select_columns("input_ids")
            .to_pandas()["input_ids"]  # Explicitly access the column
            .apply(len)
            .values
        )
```

I don't know if this is the fix.

### Which Operating Systems are you using?

- [x] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.10

### axolotl branch-commit

826fb14

### Acknowledgements

- [x] My issue title is concise, descriptive, and in title casing.
- [x] I have searched the existing issues to make sure this bug has not been reported yet.
- [x] I am using the latest version of axolotl.
- [x] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by seungduk-yanolja** [comment]
Oh, the fix worked!
